jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -711,4 +785,75 @@ class ServerKernelManager(AsyncIOLoopKernelManager):
         None, allow_none=True, help="The current execution state of the kernel"
     )
     reason = Unicode("", help="The reason for the last failure against the kernel")
+
     last_activity = Instance(datetime, help="The last activity on the kernel")
+
+    # A list of pathlib objects, each pointing at an event
+    # schema to register with this kernel manager's eventlogger.
+    # This trait should not be overridden.
+    @property
+    def core_event_schema_paths(self) -> t.List[pathlib.Path]:
+        return [DEFAULT_EVENTS_SCHEMA_PATH / "kernel_actions" / "v1.yaml"]
+
+    # This trait is intended for subclasses to override and define
+    # custom event schemas.
+    extra_event_schema_paths = List(
+        default_value=[],
+        help="""
+        A list of pathlib.Path objects pointing at to register with
+        the kernel manager's eventlogger.
+        """,
+    ).tag(config=True)
+
+    event_logger = Instance(EventLogger)
+
+    @default("event_logger")
+    def _default_event_logger(self):
+        """Initialize the logger and ensure all required events are present."""
+        if self.parent is not None and hasattr(self.parent, "event_logger"):
```

Comment:
@Zsailer - because the event_logger trait is on the `ServerKernelManager` and not the `AsyncMappingKernelManager`, do we need the _double-hop_ of parent to access the configurable on `ServerApp` here?  I.e.,:
```suggestion
        if self.parent is not None and self.parent.parent is not None and hasattr(self.parent.parent, "event_logger"):
```